### PR TITLE
Add variable properties support to fluid_scheme

### PR DESCRIPTION
**Pull request by timofeymukha**
Adds a `mu_field` to `fluid_scheme` and routine to update it based on a turbulence viscosity field.

The idea of putting it in the base type is the whole of eventually fusing full and simplified stress formulations into one solver.

There are also a bunch of changes to make flint happier.

### CODE REVIEW

File: src/fluid/fluid_pnpn.f90 [hunk 1]
```diff
@@ -71,6 +71,7 @@ module fluid_pnpn
   use mathops, only : opadd2cm, opcolv
   use bc, only: bc_list_t, bc_list_init, bc_list_add, bc_list_free, &
                 bc_list_apply_scalar, bc_list_apply_vector
+  use utils, only : neko_error
```
Comment (njansson) [inline src/fluid/fluid_pnpn.f90:74]:
```suggestion
  use utils, only : neko_error
  use field_math, only : field_add2
```

File: src/fluid/fluid_pnpn.f90 [hunk 2]
```diff
@@ -616,9 +628,9 @@ subroutine fluid_pnpn_step(this, t, tstep, dt, ext_bdf, dt_controller)
                                  this%bclst_dp, gs_Xh, n, tstep, dt_controller)
 
       if (NEKO_BCKND_DEVICE .eq. 1) then
-         call device_add2(p%x_d, dp%x_d,n)
+         call device_add2(p%x_d, dp%x_d, n)
```
Comment (njansson) [inline src/fluid/fluid_pnpn.f90:631]:
```suggestion
      call field_add2(p, dp, n)
```

File: src/fluid/fluid_pnpn.f90 [hunk 3]
```diff
@@ -616,9 +628,9 @@ subroutine fluid_pnpn_step(this, t, tstep, dt, ext_bdf, dt_controller)
                                  this%bclst_dp, gs_Xh, n, tstep, dt_controller)
 
       if (NEKO_BCKND_DEVICE .eq. 1) then
```
Comment (njansson) [inline src/fluid/fluid_pnpn.f90:630]:
```suggestion
```

File: src/fluid/fluid_pnpn.f90 [hunk 4]
```diff
@@ -616,9 +628,9 @@ subroutine fluid_pnpn_step(this, t, tstep, dt, ext_bdf, dt_controller)
...
-         call device_add2(p%x_d, dp%x_d,n)
+         call device_add2(p%x_d, dp%x_d, n)
       else
-         call add2(p%x, dp%x,n)
+         call add2(p%x, dp%x, n)
```
Comment (njansson) [inline src/fluid/fluid_pnpn.f90:633]:
```suggestion
```

File: src/fluid/fluid_pnpn.f90 [hunk 5]
```diff
@@ -616,9 +628,9 @@ subroutine fluid_pnpn_step(this, t, tstep, dt, ext_bdf, dt_controller)
...
       end if
```
Comment (njansson) [inline src/fluid/fluid_pnpn.f90:634]:
```suggestion
```

**Review by njansson — COMMENTED** [review]
🍻 Looks good. Some Minor suggestions for using the new field_math things

**Comment by timofeymukha** [comment]
@timfelle I had to update flint to stop complaining about the number of lines in a subroutine.

**Comment by timfelle** [comment]
> @timfelle I had to update flint to stop complaining about the number of lines in a subroutine.

Sure, there is a bunch of those rules that i didn't really look at. Also the one with number of arguments to functions and subroutines..

**Review by njansson — APPROVED** [review]

File: src/fluid/fluid_scheme.f90 [hunk 6]
```diff
@@ -132,27 +134,47 @@ module fluid_scheme
...
      procedure, pass(this) :: set_usr_inflow => fluid_scheme_set_usr_inflow
+     !> Compute the CFL number
      procedure, pass(this) :: compute_cfl => fluid_compute_cfl
+     !> Constructor
      procedure(fluid_scheme_init_intrf), pass(this), deferred :: init
+     !> Destructor
      procedure(fluid_scheme_free_intrf), pass(this), deferred :: free
+     !> Advance one step in time
      procedure(fluid_scheme_step_intrf), pass(this), deferred :: step
+     !> Restart from a checkpoint
```
Comment (vbaconnet) [inline src/fluid/fluid_scheme.f90:171]: @timofeymukha Where did we land on using periods at the end of docstrings? ;)

**Review by Shiyu-Sandy-Du — APPROVED** [review]

File: src/fluid/fluid_scheme.f90 (hunk 6 above)
Comment (timofeymukha) [reply src/fluid/fluid_scheme.f90:171]: I think it remains an artistic freedom :-). But I hope you are in the "using periods" camp!

**Pull request merged**